R/btw_client.R

```diff
@@ -519,6 +532,44 @@ btw_client_config <- function(client = NULL, tools = NULL, config = list()) {
   config
 }
 
+flatten_and_check_tools <- function(tools) {
+  if (isFALSE(tools)) {
+    return(list())
+  }
+
+  if (inherits(tools, "ellmer::ToolDef")) {
+    tools <- list(tools)
+  }
```

Comment:
fwiw, ellmer and acquaint error in what feels like the analogous case, noting that the user might want to wrap `tools` in `list()`.

https://github.com/tidyverse/ellmer/blob/8d038db5f5e416ac1b8728aeceb4adadace254ae/R/chat.R#L410-L412